gateway/src/apicast/policy/3scale_batcher/keys_helper.lua

```diff
@@ -0,0 +1,78 @@
+local ipairs = ipairs
+local format = string.format
+local insert = table.insert
+local concat = table.concat
+local sort = table.sort
+local unpack = table.unpack
+local ngx_re = ngx.re
+
+local _M = {}
+
+local function creds_part_in_key(creds)
+  if creds.app_id and creds.app_key then
+    return format("app_id:%s,app_key:%s", creds.app_id, creds.app_key)
+  elseif creds.user_key then
+    return format("user_key:%s", creds.user_key)
+  elseif creds.access_token then
+    return format("access_token:%s", creds.access_token)
+  end
+end
+
+local function metrics_part_in_key(usage)
+  local usages = {}
+
+  local deltas = usage.deltas
+
+  -- Need to sort the metrics. Otherwise, same metrics but in different order,
+  -- would end up in a different key.
+  local metrics = { unpack(usage.metrics) } -- Does not modify the original.
```

Comment:
Pity we can't use https://github.com/openresty/luajit2/commit/617f118303b897bf23129076eab5167c143cc568 yet.